deepmd/fit/ener.py

```diff
@@ -645,7 +676,7 @@ def build(
         self.atom_ener_before = outs * atype_filter
         self.add_type = tf.reshape(
             tf.nn.embedding_lookup(self.t_bias_atom_e, self.atype_nloc),
-            [tf.shape(inputs)[0], natoms[0]],
+            [tf.shape(inputs)[0], tf.reduce_sum(natoms[2 : 2 + ntypes_atom])],
```

Comment:
`natoms[0]` also appears in line 687 and line 691 after this line.

It seems to me that one can directly pass `tf.reduce_sum(natoms[2 : 2 + ntypes_atom])` to `natoms[0]` when preparing the feed dictionary. Is there any problem?

Reply:
For the case with spin, `tf.reduce_sum(natoms[2 : 2 + ntypes_atom])` only counts real atoms; For the case without spin, `tf.reduce_sum(natoms[2 : 2 + ntypes_atom])` is equal to `natoms[0]`.